app/controllers/computacenter/school_changes_controller.rb

```diff
@@ -0,0 +1,89 @@
+class Computacenter::SchoolChangesController < Computacenter::BaseController
+  before_action :set_school, except: :index
+
+  def index
+    respond_to do |format|
+      format.html do
+        @schools = fetch_schools
+        @show_download_link = show_download_link?
+      end
+      format.csv { send_data csv_generator, filename: "school-changes-#{Time.zone.now.strftime('%Y%m%d')}.csv" }
+    end
+  end
+
+  def edit
+    @form = Computacenter::ShipToForm.new(school: @school,
+                                          ship_to: @school.computacenter_reference)
+  end
+
+  def update
+    @form = Computacenter::ShipToForm.new(ship_to_params.merge(school: @school))
+
+    if @form.valid?
+      update_ship_to
+      flash[:success] = t(:success, scope: %i[computacenter ship_to update], name: @school.name, ship_to: @school.computacenter_reference)
+      redirect_to computacenter_school_changes_path
+    else
+      render :edit, status: :unprocessable_entity
+    end
+  end
+
+private
+
+  def set_school
+    @school = School.gias_status_open.find_by(urn: params[:id])
+  end
+
+  def csv_generator
+    SchoolDataExporter.new.export_schools(fetch_schools)
+  end
+
+  def fetch_schools
+    School.joins(:responsible_body)
+      .includes(:responsible_body)
+      .gias_status_open
+      .merge(query_for_view_mode)
+      .order(ResponsibleBody.arel_table[:type].asc, ResponsibleBody.arel_table[:name].asc, School.arel_table[:name].asc)
+  end
```

Comment:
Is there a way to push this method (and maybe `query_for_view_mode`)  onto the `School` model? It feels too domain-logicy to be on the controller...